File: Application/Sources/UI/Views/TableLoadMoreFooterView.m
//
//  Copyright (c) SRG SSR. All rights reserved.
//
//  License information is available from the LICENSE file.
//

#import "TableLoadMoreFooterView.h"

#import "UIImageView+PlaySRG.h"

@import SRGAppearance;

@implementation TableLoadMoreFooterView

- (instancetype)initWithFrame:(CGRect)frame
{
    if (self = [super initWithFrame:frame]) {
        self.backgroundColor = UIColor.clearColor;
        
        UIImageView *loadingImageView = [UIImageView play_loadingImageViewWithTintColor:UIColor.srg_grayC7Color];
        [self addSubview:loadingImageView];
        
        loadingImageView.translatesAutoresizingMaskIntoConstraints = NO;
        [NSLayoutConstraint activateConstraints:@[
            [loadingImageView.centerXAnchor constraintEqualToAnchor:self.centerXAnchor],
            [loadingImageView.centerYAnchor constraintEqualToAnchor:self.centerYAnchor],
        ]];
    }
    return self;
}

@end

File: Application/Sources/UI/Views/TableLoadMoreFooterView.swift
//
//  Copyright (c) SRG SSR. All rights reserved.
//
//  License information is available from the LICENSE file.
//

import SRGAppearance
import UIKit

class TableLoadMoreFooterView: UIView {
    override init(frame: CGRect) {
        super.init(frame: frame)

        backgroundColor = .clear

        let loadingImageView = UIImageView.play_loadingImageView(withTintColor: .srgGrayD2)
        addSubview(loadingImageView)

        loadingImageView.translatesAutoresizingMaskIntoConstraints = false
        NSLayoutConstraint.activate([
            loadingImageView.centerXAnchor.constraint(equalTo: centerXAnchor),
            loadingImageView.centerYAnchor.constraint(equalTo: centerYAnchor)
        ])
    }

    required init?(coder aDecoder: NSCoder) {
        super.init(coder: aDecoder)
    }
}
